packages/typespec-client-generator-core/src/types.ts

```diff
@@ -531,6 +536,54 @@ export function getSdkEnum(context: SdkContext, type: Enum, operation?: Operatio
   return sdkType;
 }
 
+function getSdkUnionEnumValues(
+  context: SdkContext,
+  type: UnionEnum,
+  enumType: SdkEnumType
+): SdkEnumValueType[] {
+  const values: SdkEnumValueType[] = [];
+  for (const [name, member] of type.flattenedMembers.entries()) {
+    const docWrapper = getDocHelper(context, member.variant);
+    values.push({
+      kind: "enumvalue",
+      name: typeof name === "string" ? name : `${member.value}`,
+      description: docWrapper.description,
+      details: docWrapper.details,
+      value: member.value,
+      valueType: enumType.valueType,
+      enumType,
+      nullable: false,
+    });
+  }
+  return values;
+}
+
+function getSdkUnionEnum(context: SdkContext, type: UnionEnum, operation?: Operation) {
+  let sdkType = context.modelsMap?.get(type.union) as SdkEnumType | undefined;
+  if (!sdkType) {
+    if (!type.union.name) throw new Error("Your union must be named in order to generate an enum");
+    const union = type.union as Union & { name: string };
+    const docWrapper = getDocHelper(context, union);
+    sdkType = {
+      ...getSdkTypeBaseHelper(context, type.union, "enum"),
+      name: getLibraryName(context, type.union),
+      description: docWrapper.description,
+      details: docWrapper.details,
+      valueType: { ...getSdkTypeBaseHelper(context, type.kind, "string"), encode: "string" },
+      values: [],
+      nullable: false,
+      isFixed: !type.open,
+      isFlags: false,
+      usage: UsageFlags.None, // We will add usage as we loop through the operations
```

Comment:
Can we leave this as undefined until it's set later instead of setting it to `None`?

Reply:
`usage` is a required value so we'll have a compile error if we set it to `undefined`